app/ide-desktop/lib/client/src/file-associations.ts

```diff
@@ -0,0 +1,65 @@
+import pathModule from 'node:path'
+import * as paths from './paths'
+import * as electron from 'electron'
+import child_process from 'node:child_process'
+import process from 'node:process'
+import { BUNDLED_PROJECT_EXTENSION, SOURCE_FILE_EXTENSION } from '../file-associations'
+import * as project from './project-management'
+import { logger } from 'enso-content-config'
+import { PRODUCT_NAME } from 'enso-common'
+
+export * from '../file-associations'
+
+/* Check if the given path looks like a file that we can open. */
+export function isFileOpenable(path: string): boolean {
+    const extension = pathModule.extname(path).toLowerCase()
+    return extension === BUNDLED_PROJECT_EXTENSION || extension === SOURCE_FILE_EXTENSION
+}
+
+/* On macOS when Enso-associated file is opened, the application is first started and then it
+ * receives the `open-file` event. However, if there is already an instance of Enso running,
+ * it receives the `open-file` event (and no new instance is created for us). In this case,
+ * we manually start a new instance of the application and pass the file path to it (using the
+ * Windows-style command).
+ */
+export function onFileOpened(event: Event, path: string) {
+    if (isFileOpenable(path)) {
+        // If we are not ready, we can still decide to open a project rather than enter the welcome
+        // screen. However, we still check for the presence of arguments, to prevent hijacking the
+        // user-spawned IDE instance (OS-spawned will not have arguments set).
+        if (!electron.app.isReady() && paths.clientArguments.length === 0) {
+            event.preventDefault()
+            logger.log(`Opening file '${path}'.`)
+            return handleOpenFile(path)
+        } else {
+            // We need to start another copy of the application, as the first one is already running.
+            logger.log(`We are already initialized, opening '${path}' in a new instance.`)
```

Comment:
Very minot `We are ...` - `The app is ...` - this is just a little bit more formal.

Reply:
Fixed.